src/v/iceberg/partition_json.cc

```diff
@@ -0,0 +1,84 @@
+// Copyright 2024 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+#include "iceberg/partition_json.h"
+
+#include "iceberg/json_utils.h"
+#include "iceberg/partition.h"
+#include "iceberg/transform_json.h"
+#include "json/document.h"
+
+#include <stdexcept>
+
+namespace iceberg {
+
+partition_field parse_partition_field(const json::Value& v) {
+    auto source_id = parse_required_i32(v, "source-id");
+    auto field_id = parse_required_i32(v, "field-id");
+    auto name = parse_required_str(v, "name");
+    auto transform_str = parse_required_str(v, "transform");
+    auto transform = transform_from_str(transform_str);
+    return partition_field{
+      .source_id = nested_field::id_t{source_id},
+      .field_id = partition_field::id_t{field_id},
+      .name = name,
+      .transform = transform,
+    };
+}
+
+partition_spec parse_partition_spec(const json::Value& v) {
+    auto spec_id = parse_required_i32(v, "spec-id");
+    const auto& fields_array = parse_required(v, "fields");
+    if (!fields_array.IsArray()) {
+        throw std::invalid_argument(fmt::format(
+          "Expected 'fields' to be array: {}", fields_array.GetType()));
+    }
+    chunked_vector<partition_field> fs;
```

Comment:
nit: reserve for size of `fields_array`? does that make sense for chunked vector?

Reply:
Done